crates/bevy_render/src/camera/projection.rs

```diff
@@ -208,34 +225,16 @@ impl CameraProjection for OrthographicProjection {
             ScalingMode::FixedHorizontal(viewport_width) => {
                 (viewport_width, height * viewport_width / width)
             }
-            ScalingMode::None => return,
+            ScalingMode::None { width, height } => (width, height),
         };
 
-        match self.window_origin {
-            WindowOrigin::Center => {
-                let half_width = viewport_width / 2.0;
-                let half_height = viewport_height / 2.0;
-                self.left = -half_width;
-                self.bottom = -half_height;
-                self.right = half_width;
-                self.top = half_height;
+        let origin_x = frustum_width * self.viewport_origin.0;
+        let origin_y = frustum_height * self.viewport_origin.1;
 
-                if let ScalingMode::WindowSize = self.scaling_mode {
-                    if self.scale == 1.0 {
-                        self.left = self.left.floor();
-                        self.bottom = self.bottom.floor();
-                        self.right = self.right.floor();
-                        self.top = self.top.floor();
-                    }
-                }
-            }
-            WindowOrigin::BottomLeft => {
-                self.left = 0.0;
-                self.bottom = 0.0;
-                self.right = viewport_width;
-                self.top = viewport_height;
-            }
-        }
+        self.left = -origin_x * self.scale;
```

Comment:
(this is not a review comment, just a request for explanation)

Why is there a negative sign in: `left = -origin_x * self.scale;`?

Reply:
`origin_x` is the distance from the viewport's left side to the origin.

`self.left` is the distance from the origin to the frustum's left side. I.e. opposite of `origin_x`.